src/main/kotlin/uk/gov/justice/digital/hmpps/hmppschallengesupportinterventionplanapi/service/CsipRecordService.kt

```diff
@@ -0,0 +1,60 @@
+package uk.gov.justice.digital.hmpps.hmppschallengesupportinterventionplanapi.service
+
+import jakarta.transaction.Transactional
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppschallengesupportinterventionplanapi.client.prisonersearch.PrisonerSearchClient
+import uk.gov.justice.digital.hmpps.hmppschallengesupportinterventionplanapi.config.CsipRequestContext
+import uk.gov.justice.digital.hmpps.hmppschallengesupportinterventionplanapi.domain.toCsipRecord
+import uk.gov.justice.digital.hmpps.hmppschallengesupportinterventionplanapi.domain.toModel
+import uk.gov.justice.digital.hmpps.hmppschallengesupportinterventionplanapi.enumeration.ReferenceDataType.AREA_OF_WORK
+import uk.gov.justice.digital.hmpps.hmppschallengesupportinterventionplanapi.enumeration.ReferenceDataType.CONTRIBUTORY_FACTOR_TYPE
+import uk.gov.justice.digital.hmpps.hmppschallengesupportinterventionplanapi.enumeration.ReferenceDataType.INCIDENT_INVOLVEMENT
+import uk.gov.justice.digital.hmpps.hmppschallengesupportinterventionplanapi.enumeration.ReferenceDataType.INCIDENT_LOCATION
+import uk.gov.justice.digital.hmpps.hmppschallengesupportinterventionplanapi.enumeration.ReferenceDataType.INCIDENT_TYPE
+import uk.gov.justice.digital.hmpps.hmppschallengesupportinterventionplanapi.model.CsipRecord
+import uk.gov.justice.digital.hmpps.hmppschallengesupportinterventionplanapi.model.request.CreateCsipRecordRequest
+import uk.gov.justice.digital.hmpps.hmppschallengesupportinterventionplanapi.repository.CsipRecordRepository
+import uk.gov.justice.digital.hmpps.hmppschallengesupportinterventionplanapi.repository.ReferenceDataRepository
+
+@Service
+class CsipRecordService(
+  val referenceDataRepository: ReferenceDataRepository,
+  val csipRecordRepository: CsipRecordRepository,
+  val prisonerSearchClient: PrisonerSearchClient,
+) {
+
+  @Transactional
+  fun createCsipRecord(
+    request: CreateCsipRecordRequest,
+    prisonNumber: String,
+    requestContext: CsipRequestContext,
+  ): CsipRecord {
+    val prisoner = prisonerSearchClient.getPrisoner(prisonNumber)
+    require(prisoner != null) { "Prisoner with prison number $prisonNumber could not be found" }
+    val incidentType = referenceDataRepository.findByDomainAndCode(INCIDENT_TYPE, request.referral.incidentTypeCode)
+    val incidentLocation =
+      referenceDataRepository.findByDomainAndCode(INCIDENT_LOCATION, request.referral.incidentLocationCode)
+    val referrerAreaOfWork = referenceDataRepository.findByDomainAndCode(AREA_OF_WORK, request.referral.refererAreaCode)
+    val incidentInvolvement =
+      referenceDataRepository.findByDomainAndCode(INCIDENT_INVOLVEMENT, request.referral.incidentInvolvementCode)
+    require(incidentType != null) { "Incident type code ${request.referral.incidentTypeCode} could not be found" }
```

Comment:
These check for missing codes but don't check for inactive ones

Reply:
Good spot. You can solve this a number of ways - I suppose do we need to know if it exists but isn't active, or should it return null?